### PR TITLE
docs: rm type hints in ignite.distributed (5)

### LINKED ISSUE #1665: Use autodoc_typehints = "description" in conf.py
## 📚 Documentation

EDITED HERE: As a help wanted issue, please, see below comment https://github.com/pytorch/ignite/issues/1665#issuecomment-783240506 on what remains to do here. Thanks 

If there are many arguments in class/function/method, it's a little hard to read/scan all arguments (due to typehints).
Although there is a `Parameters` section, there might be missing some arguments (rarely though).
And we are duplicating type hints work in both signatures and docstring. Writing once can make less work on authors and less type hints mismatch can occur.

If we can remove type hints, it can be a little easier to read, and showing type hints only in `Parameters` section.

For example:
Before:
Screen Shot 2021-02-21 at 22 52 16
After:
Screen Shot 2021-02-21 at 22 51 40

Changing to this docs style needs a little work. It needs to 
- add
```py
autoclass_content = "both"
autodoc_typehints = "description"  # new in sphinx 3.1
```
those lines in `conf.py`
- removing type hints in docstring
- leaving type hints only in signatures.

Let me know what you think.
cc: @vfdev-5 @sdesrozis @trsvchn

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1665 
Fixes #1665 
https://deploy-preview-1687--pytorch-ignite-preview.netlify.app/distributed.html

Description:
- remove type hints in ignite.distributed
- add missing doc string

Check list:

- [ ] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-sphinx==3.2.1
+sphinx==3.4
```
Comment (vfdev-5) [inline docs/requirements.txt:1]: Maybe, we can split this PR into multiple ones to simplify the review and also 
let's have sphinx version update in a separate PR.

File: ignite/metrics/accumulation.py [hunk 2]
```diff
@@ -25,13 +25,13 @@ class VariableAccumulation(Metric):
         - `+batch_size` if input is a ND `torch.Tensor`. Batch size is the first dimension (`shape[0]`).
```
Comment (vfdev-5) [inline ignite/metrics/accumulation.py:25]: Let's remove all files related to metrics package :)

Comment (ydcjeff) [reply ignite/metrics/accumulation.py:25]: I am expecting `ignite.metrics` to merge first, then this
But, ok I will remove metrics files here

File: docs/source/conf.py [hunk 3]
```diff
@@ -209,7 +209,6 @@
 autodoc_inherit_docstrings = True
 autoclass_content = "both"
 autodoc_typehints = "description"
-napoleon_attr_annotations = True
```
Comment (vfdev-5) [inline docs/source/conf.py]: Shouldn't we keep this ?

**Review by vfdev-5 — APPROVED** [review]
LGTM! Thanks @ydcjeff !
I think we can merge this one before metrics, right ?

**Comment by ydcjeff** [comment]
> LGTM! Thanks @ydcjeff !
> I think we can merge this one before metrics, right ?

Yes, both can be merge independently now.

**Pull request merged**